matter_server/server/device_controller.py

```diff
@@ -99,12 +109,16 @@ async def initialize(self) -> None:
     async def start(self) -> None:
         """Handle logic on controller start."""
         # load nodes from persistent storage
-        nodes: dict[str, dict] = self.server.storage.get(DATA_KEY_NODES, {})
+        nodes: dict[str, dict | None] = self.server.storage.get(DATA_KEY_NODES, {})
         for node_id_str, node_dict in nodes.items():
             node_id = int(node_id_str)
-            # invalidate node data if schema mismatch,
-            # the node will automatically be scheduled for re-interview
-            if node_dict and node_dict.get("interview_version") != SCHEMA_VERSION:
+            if node_dict is None:
+                # ignore non-initialized (left-over) nodes
+                # from failed commissioning attempts
+                continue
```

Comment:
IMHO we should still retry to interview those. It could be that the user commissioned them, then interviewing failed, he restarts the add-on and now maybe this second time interviewing works... :man_shrugging: 

I understand that now people have probably such stale entries, and this code makes it retry every time. And since people don't see those in Core, they also cant' delete them.

So, as a work around, I think this is fine.

But once we should them, have them cleaned-up, we should no longer ignore them.

Reply:
No, we agreed on this. This is old stuff. It can not happen anymore but we need to clean out the old traces.